cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```
@@ -279,6 +293,64 @@ public SparkConfig generateSparkSubmitConfig(SparkSubmitContext sparkSubmitConte
                            sparkConfMap);
   }
 
+  @Override
+  public void onNamespaceCreation(String cdapNamespace, Map<String, String> properties) throws Exception {
+    String namespace = properties.get(NAMESPACE_PROPERTY);
+    if (StringUtils.isEmpty(namespace)) {
+      LOG.info("Kubernetes namespace not created for {} because no name was provided in the config", cdapNamespace);
+      return;
+    }
+    try {
+      V1NamespaceList namespaceList = coreV1Api.listNamespace(null, null, null,
+              String.format("metadata.name=%s", namespace), null, 1, null, null, 10, false);
+      if (!namespaceList.getItems().isEmpty()) {
+        throw new IOException(String.format("Kubernetes namespace %s already exists", namespace));
+      }
+    } catch (ApiException e) {
+      throw new IOException("Cannot check if Kubernetes namespace already exists. Error code = "
+              + e.getCode() + ", Body = " + e.getResponseBody(), e);
+    }
+
+    V1Namespace namespaceObject = new V1Namespace();
+    namespaceObject.setMetadata(new V1ObjectMeta().name(namespace).putLabelsItem(CDAP_NAMESPACE_LABEL, cdapNamespace));
+
+    String kubeCpuLimit = properties.get(NAMESPACE_CPU_LIMIT_PROPERTY);
+    String kubeMemoryLimit = properties.get(NAMESPACE_MEMORY_LIMIT_PROPERTY);
+    Map<String, Quantity> hardLimitMap = new HashMap<>();
+    if (StringUtils.isNotEmpty(kubeCpuLimit)) {
```

Use `Quantity.parse` instead. It handles `null` and empty string.

---

`parse` will throw a `QuantityFormatException` if the value is null or empty. Since the properties are optional, I think it'd be better to do the null/empty check instead of catching and ignoring the exception